src/store/reducers/rulesReducer.js

```diff
@@ -0,0 +1,131 @@
+            ])
+            .startWith(redux.actions.updateRules(rules, { fromAction }))
+        )
+        .catch(handleError(fromAction))
+  },
+
+  fetchRuleCounts: {
+    type: 'RULES_COUNT_FETCH',
+    epic: (fromAction, store) =>
+      Observable.of(getDeviceGroupEntities(store.getState()))
+        .map(entities => entities[fromAction.payload.groupId])
+        .flatMap(({ conditions }) => IoTHubManagerService.getDevices(conditions))
+        .map(devices => devices.length)
+        .map(count => redux.actions.updateRuleCount({ id: fromAction.payload.id, count }))
+        .catch(handleError(fromAction)) // Needs to work with selector
+  },
+
+  fetchRuleLastTriggered: {
+    type: 'RULES_LAST_TRIGGER_FETCH',
+    epic: (fromAction, store) =>
+      TelemetryService.getAlarmsByRule(fromAction.payload, {
+        order: 'desc',
+        limit: 1,
+        devices: getDeviceIds(store.getState()).join(',')
+      })
+      .map(([alarm]) => redux.actions.updateRuleLastTrigger({ id: fromAction.payload, lastTrigger: alarm.dateModified }))
+      .catch(handleError(fromAction)) // Needs to work with selector
+  }
+});
+// ========================= Epics - END
+
+// ========================= Schemas - START
+const ruleSchema = new schema.Entity('rules');
+const ruleListSchema = new schema.Array(ruleSchema);
+// ========================= Schemas - END
+
+// ========================= Reducers - START
+const initialState = { ...errorPendingInitialState, entities: {}, items: [] };
+
+const updateRulesReducer = (state, { payload, fromAction }) => {
+  const { entities: { rules }, result } = normalize(payload, ruleListSchema);
+  return update(state, {
+    entities: { $set: rules },
+    items: { $set: result },
+    lastUpdated: { $set: moment() },
+    ...setPending(fromAction.type, false)
+  });
+};
+
+const updateCountReducer = (state, { payload: { id, count } }) => update(state, {
+  entities: { [id]: { count: { $set: count } } }
+});
+
+const updateLastTriggerReducer = (state, { payload: { id, lastTrigger } }) => update(state, {
+  entities: { [id]: { lastTrigger: { $set: lastTrigger } } }
+});
+
+/* Action types that cause a pending flag */
+const fetchableTypes = [
+  epics.actionTypes.fetchRules
+];
+
+export const redux = createReducerScenario({
+  updateRules: { type: 'RULES_UPDATE', reducer: updateRulesReducer },
+  updateRuleCount: { type: 'RULES_COUNT_UPDATE', reducer: updateCountReducer },
+  updateRuleLastTrigger: { type: 'RULES_LAST_TRIGGER_UPDATE', reducer: updateLastTriggerReducer },
+  registerError: { type: 'RULES_REDUCER_ERROR', reducer: errorReducer },
+  isFetching: { multiType: fetchableTypes, reducer: pendingReducer },
+});
+
+export const reducer = { rules: redux.getReducer(initialState) };
+// ========================= Reducers - END
+
+// ========================= Selectors - START
+export const getRulesReducer = state => state.rules;
+export const getEntities = state => getRulesReducer(state).entities;
```

Comment:
Same comment as I made for getEntities/getItems for devices.

Reply:
See above